### PR TITLE
integrate @qlik/sdk for authentication

### PR DESCRIPTION
**Pull request by a-m-dev**
Items in this PR: 
- [x] creating a base class for OAuth
- [x] implementing connect with `webIntegrationId`
- [x] implementing connect with csrf-token (refactor for old codes)

### CODE REVIEW

File: examples/simple-setup/src/index.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import "./styles.css";
 import embed from "./configure";
 import connect from "./connect";
-
+import { CONNECTION_TYPES } from "./connect";
```
Comment (LiKang6688) [inline examples/simple-setup/src/index.js:4]:
```suggestion
import connect, { CONNECTION_TYPES } from "./connect";
```
and remove the code one line above

**Comment by LiKang6688** [comment]
It would be good to add a readme and qlik.dev link about the @qlik/sdk and authentication

**Comment by a-m-dev** [comment]
will add the info for readme but no idea about qlik.dev

**Comment by LiKang6688** [comment]
> will add the info for readme but no idea about qlik.dev

is it this one https://qlik.dev/libraries-and-tools/platform-sdk#authentication?

**Pull request closed without merging**